docs/2.learn/2.concepts.md

```
@@ -255,6 +256,70 @@ results, err := index.Query(ctx, queryVector, 10, filter)
 
 **Note:** Filter operator support varies by provider. See [Provider Reference](../5.reference/2.providers.md) for the operator support matrix.
 
+## Lifecycle Hooks
+
+Types can opt into lifecycle hooks by implementing one or more hook interfaces. When present, grub calls these hooks at specific points during CRUD operations across all four storage types.
+
+### Hook Interfaces
+
+| Interface | When Called | Effect of Error |
+|-----------|-----------|-----------------|
```

markdownlint-cli2 (0.20.0)

Table column style
Table pipe is missing space to the right for style "compact"

(MD060, table-column-style)